app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -1591,6 +1605,344 @@ class StateFragmentTest {
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_differentTextBaseInteractions_doesNotShareInitialState() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      // Entering text in Fraction Input Interaction
+      typeFractionText("1/2")
+      // Rotating device
+      rotateToLandscape()
+      it.onActivity {
+        val fractionInputInteraction =
+          it.findViewById<FractionInputInteractionView>(R.id.fraction_input_interaction_view)
+        assertThat(fractionInputInteraction.text.toString()).isEqualTo("1/2")
+      }
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+      playThroughPrototypeState3()
+      playThroughPrototypeState4()
+      playThroughPrototypeState5()
+
+      it.onActivity {
+        val numericInputInteractionView =
+          it.findViewById<NumericInputInteractionView>(R.id.numeric_input_interaction_view)
+        assertThat(numericInputInteractionView.text.toString().isEmpty()).isTrue()
+      }
+    }
+  }
+
+  @Test // TODO(#4692): Robolectric tests not working on screen rotation for input interactions
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_checkPreviousHeaderExpanded_retainStateOnConfigChange() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/4")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit a second wrong answer.
+      typeFractionText("1/4")
+      clickSubmitAnswerButton()
+
+      // Expand previous response header.
+      scrollToViewType(PREVIOUS_RESPONSES_HEADER)
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.previous_response_header)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // rotate screen
+      rotateToLandscape()
+
+      // Both failed answers should be showing.
+      onView(withId(R.id.state_recycler_view))
+        .check(
+          matchesChildren(matcher = withId(R.id.submitted_answer_container), times = 2)
+        )
+    }
+  }
+
+  @Test // TODO(#4692): Robolectric tests not working on screen rotation for input interactions
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_checkPreviousHeaderCollapsed_retainStateOnConfigChange() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/4")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit a second wrong answer.
+      typeFractionText("1/4")
+      clickSubmitAnswerButton()
+
+      // rotate screen (by default responses are collapsed)
```

Comment:
```suggestion
      // Rotate screen (by default responses are collapsed).
```

For consistency. Ditto elsewhere in this test suite (I think there are other cases).

Reply:
Updated